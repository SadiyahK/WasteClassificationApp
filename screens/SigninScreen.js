/**
 * SigninScreen: Handles user authentication of existing users.
 */
import React, { Component } from 'react';
import {Text, View, TextInput, Alert, ActivityIndicator, TouchableOpacity, Image } from 'react-native';
import firebase from '../database/firebase';
import stylesheet from '../styles/stylesheet.js'

export default class SigninScreen extends Component {
  
  constructor() {
    super();
    this.state = { 
      email: '', 
      isLoading: false,
      password: ''
    }
  }

  // Take each input from each field and store
  // it in the appropriate placeholder.
  onTextInput = (textInput, prop) => {
    const state = this.state;
    state[prop] = textInput;
    this.setState(state);
  }

  // Handles communication with Firebase to 
  // confirm user details and navigate to profile.
  onSignInClick = () => {
    if(this.state.email === '' || this.state.password === '') {
      Alert.alert('Enter details to sign in!')
    } else {
      this.setState({ isLoading: true })
      //firebase communication
      firebase.auth().signInWithEmailAndPassword(this.state.email, this.state.password)
      .then((res) => {
        console.log('sign in successful')
        this.setState({ email: '', password: '', isLoading: false })
        this.props.navigation.navigate('Profile') // navigate to user's profile
      })
      .catch(error => Alert.alert('An error occurred. Please try again later.'))
    }
  }

  render() {
    // If authentication is in progress...
    if(this.state.isLoading){
      return(
        <View style={stylesheet.preloader}>
          <ActivityIndicator size="large" color="#9E9E9E"/>
        </View>
      )
    } 

    // Otherwise, show sign in
    return (
        <View>
          {/* Top Icon */}
          <View style={stylesheet.topContainer}>
            <Image source={require('../assets/p-trans.png')} style={stylesheet.imageIcon}/>
          </View>
          {/* Input fields */}
          <View style={stylesheet.container}>
            <TextInput
              style={stylesheet.inputStyle}
              placeholder="Email"
              placeholderTextColor= '#fff'
              value={this.state.email}
              onChangeText={(val) => this.onTextInput(val, 'email')}
            />
            <TextInput
              style={stylesheet.inputStyle}
              placeholder="Password"
              placeholderTextColor= '#fff'
              value={this.state.password}
              onChangeText={(val) => this.onTextInput(val, 'password')}
              maxLength={15}
              secureTextEntry={true}
            />
            {/* navigation options: sign in, sign up or reset password */}
            <TouchableOpacity onPress={() => this.onSignInClick()} style={stylesheet.appButtonContainer}>
              <Text style={ stylesheet.button } onPress={() => this.onSignInClick()}>Sign In</Text>
            </TouchableOpacity>
            <Text style={stylesheet.loginText} onPress={() => this.props.navigation.navigate('Signup')}>
                Don't have an account? Click here to sign up
            </Text>
            <Text style={stylesheet.loginText} onPress={() => this.props.navigation.navigate('ResetPassword')}>
                Forgot your password? Reset here
            </Text>                          
          </View>
      </View>
    );
  }
}