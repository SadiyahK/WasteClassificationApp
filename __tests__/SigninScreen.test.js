
import React from 'react';
import create from "react-test-renderer";
import ReactDOM from 'react-dom'
import renderer from "react-test-renderer";
//import { render, fireEvent, screen } from '@testing-library/react'
import { render } from '@testing-library/react-native'
import { shallow } from 'enzyme'

import SigninScreen from '../screens/SigninScreen';



// check placeholder text of email and password
// check alert pops up when one or more text fields is left blank
// check screen navigates to profile when input is correct and the sign in button is pressed
// check navigation to resetPass when link clicked
// check navigation to sign-up when link clicked.
{/* <input> element value should be empty (2x of these)
should update the state when a value is input
should display an error when no value is input */}

it("renders default elements", () =>{
    const { getAllByText, getByPlaceholderText } = render(<SigninScreen />)

    expect(getAllByText("Sign In").length).toBe(1)
    getByPlaceholderText("Email")
    getByPlaceholderText("Password")
})


// it('renders without crashing', () =>{
//     shallow(<SigninScreen />)
// })

// test('something'), () =>{
//     expect('something').to
// })

// const navigation = {
//     navigation: jest.fn()
// }
// const tree = renderer.create(<SigninScreen navigation={ navigation } />).toJSON();

// test('snapshot', () =>{
//     expect(tree).toMatchSnapshot()
// })

// test('navigate to signup', () =>{
//     const text = tree.root.findByProps({testID: 'sign-up-txt'}).props
//     text.onPress()
//     expect(navigation.navigate).toBeCalledWith('Signup')
// })

// test('placeholders', async () =>{
//     // const placeholder = tree.root.getByPlaceholderText('Email'); 
//     // expect(placeholder).tob
//     render(<SigninScreen />)
//     const text = screen.findByPlaceholderText("Email")
//     expect(text).toBe("Email")
// })

// describe("<App />", () => {
//     it('has 1 child', () => {
//         const tree = renderer.create(<App />).toJSON();
//         expect(tree.children.length).toBe(1);
//     });
// });