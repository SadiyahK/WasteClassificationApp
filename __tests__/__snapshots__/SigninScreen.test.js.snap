// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "35%",
        "justifyContent": "center",
        "paddingBottom": 15,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 100,
          "height": "80%",
          "width": "40%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(52, 52, 52, 0.55)",
        "borderRadius": 5,
        "flexDirection": "column",
        "height": "50%",
        "justifyContent": "center",
        "marginHorizontal": "5%",
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#fff"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderBottomWidth": 1,
          "borderColor": "#ccc",
          "color": "#fff",
          "marginBottom": 15,
          "paddingBottom": 15,
          "paddingLeft": 8,
          "paddingTop": 15,
          "textShadowColor": "#fff",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      maxLength={15}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#fff"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "center",
          "borderBottomWidth": 1,
          "borderColor": "#ccc",
          "color": "#fff",
          "marginBottom": 15,
          "paddingBottom": 15,
          "paddingLeft": 8,
          "paddingTop": 15,
          "textShadowColor": "#fff",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#8FD14F",
          "borderRadius": 5,
          "marginTop": 10,
          "opacity": 1,
          "paddingHorizontal": 12,
          "paddingVertical": 10,
          "width": "40%",
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFFEF2",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Sign In
      </Text>
    </View>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#fff",
          "marginTop": 25,
          "textAlign": "center",
        }
      }
    >
      Don't have an account? Click here to sign up
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#fff",
          "marginTop": 25,
          "textAlign": "center",
        }
      }
    >
      Forgot your password? Reset here
    </Text>
  </View>
</View>
`;
